### PR TITLE
increases page padding when subnav wraps

### PR DESCRIPTION
**Pull request by PeculiarGoat**
Bumps the whole page down by nav-item height if the subnav wraps.  Bumps it back up if the subnav unwraps.

### CODE REVIEW

File: site/static/js/main.js [hunk 1]
```diff
@@ -472,3 +507,5 @@ document.addEventListener('s72loaded', function(event) {
   let app = event.detail.app;
   documentReady(app);
 });
+
+window.onresize = togglePageTopPadding;
```
Comment (gscragg) [inline site/static/js/main.js:511]: Please use `window.addEventListener` and move it into `documentReady()`

Comment (gscragg) [reply site/static/js/main.js:511]: You should also do your best to ensure this function isn't called every cpu cycle as the user resizes.
Without any type of throttling this function will run 100s of times per resize.

The slider and - maybe - carousel have examples on throttling resize logic.

File: site/static/js/main.js [hunk 2]
```diff
@@ -427,6 +427,39 @@ function toggleMobileMenu(icon){
...
+
+function togglePageTopPadding() {
+  var navItem = document.querySelector('.sub-nav .nav-item');
+  if (!navItem) return;
+
+  var page = document.querySelector('.page');
+  if (!page) return;
+
+  if (subnavOverflowing()) {
+    if (page.hasAttribute('style')) return;
```
Comment (gscragg) [inline site/static/js/main.js:455]: This implies that we are only supporting a single resize?

**Review by gscragg — CHANGES_REQUESTED** [review]

File: site/static/js/main.js (hunk 2 above)
Comment (PeculiarGoat) [reply site/static/js/main.js:455]: This prevents the padding from growing 100s of times per resize.  I realise that this check being indirect isn't a great way.  Maybe I could hold onto the page padding value in a global variable instead?

**Review by gscragg — CHANGES_REQUESTED** [review]
See if the `s72.utils.fn.throttle()` suits your needs and if so use that instead of rolling your own.

**Review by gscragg — APPROVED** [review]

**Pull request merged**